src/zarr/core/buffer/cpu.py

```diff
@@ -154,7 +154,7 @@ def create(
         order: Literal["C", "F"] = "C",
         fill_value: Any | None = None,
     ) -> Self:
-        if fill_value is None:
+        if fill_value is None or fill_value == 0:
```

Comment:
could we get a comment here that explains why we are branching on the fill value?

Reply:
yes!